crates/assistant2/src/assistant_panel.rs

```diff
@@ -300,6 +300,16 @@ impl AssistantPanel {
     fn render_toolbar(&self, cx: &mut ViewContext<Self>) -> impl IntoElement {
         let focus_handle = self.focus_handle(cx);
 
+        let title = if self.thread.read(cx).is_empty() {
+            "New Thread".to_string()
```

Comment:
We should be using `SharedString` as the representation for the title, to avoid unnecessary string allocations.

The summary on the thread is already stored as a `SharedString` for cheap clones, and these static strings can be converted to a `SharedString` with `SharedString::from`.